flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/exception/LastReconciledSpecException.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes.operator.exception;
+
+/** Exception when last status is null. */
+public class LastReconciledSpecException extends RuntimeException {
```

Comment:
Seems like you are using exception for flow control instead of explicitly checking the last reconciled spec, can we avoid this?

Reply:
For some reason the deployment was corrupted, and could never finish starting, and it is impossible to delete it, impossible.

This happened to me several times, in different namespaces.